doc/specs/Configuration object for profiles.md

```diff
@@ -0,0 +1,94 @@
+---
+author: <Pankaj> <Bhojwani> <pabhojwa@microsoft.com>
+created on: <2020-11-20>
+last updated: <2020-11-20>
+issue id: <#8345>
+---
+
+# Configuration object for profiles
+
+## Abstract
+
+This spec outlines how we can support 'configuration objects' in our profiles, which
+will allow us to render differently depending on the state of the control. For example, a
+control can be rendered differently if its focused as compared to when its unfocused. Another
+example is that an elevated state control can be rendered differently as compared to a
+non-elevated one.
+
+## Inspiration
+
+Reference: [#3062](https://github.com/microsoft/terminal/issues/3062)
+
+Users want there to be a more visible indicator than the one we have currently for which
+pane is focused and which panes are unfocused. This change would grant us that feature.
+
+## Solution Design
+
+A new class in the `TerminalControl` namespace, called `CustomRenderConfig`, that will contain rendering
+parameters. `TermControl` and `TerminalCore` will use objects of this class to pass along rendering parameters
+to the renderer.
+
+Our `TerminalSettingsModel` will parse out that object from the settings json file and pipe it over to
+`TermControl`/`TerminalCore` to use. This will be done through `IControlSettings` and `ICoreSettings`, which
+is the way we already pipe over information that these interfaces need to know regarding rendering (such as
+`CursorStyle` and `FontFace`).
+
+### Allowed parameters
+
+Not all parameters which can be defined in a `Profile` can be defined in this new object (for example, we
+do not want parameters which would cause a resize in this object.) Here are the list of parameters we
+will allow:
+
+- Anything regarding colors: `colorScheme`, `foreground`, `background`, `cursorColor` etc
+- `cursorShape`
+- `backgroundImage`
+
+## UI/UX Design
+
+Users will be able to add a new setting to their profiles that will look like this:
+
+```
+"unfocusedState": 

```

Comment:
```suggestion
"state.unfocused": 
```
I think it'd be cool if we standardized this dot-notation in all of our settings. Makes it really nice for intellisense too because when I type "state", I know what states there are.